src/main/groovy/nextflow/processor/TaskProcessor.groovy

```diff
@@ -1793,6 +1794,19 @@ class TaskProcessor {
                     contextMap.put( param.name, val )
                     break
 
+                case FilemapInParam:
+                    def normalized = normalizeInputToFiles(val.values(),count)
+                    def mapfile = val
+                    int i = 0
+                    mapfile.keySet().each {
+                        mapfile[it] = Paths.get(normalized[i++].stageName)
+                    }
+
+                    contextMap.put( param.name,  mapfile)
+                    count += normalized.size()
+                    task.setInput(param, normalized)
+                    return
+
```

Comment:
I think this work in basic cases, but does not handle list of files of remote files (eg S3 objects). Also in principle a map of file should be a generalisation of a normal `file` type. Hence it should be managed in the same manner. 

The code should be moved in the `makeTaskContextStage2` and to each entry in the map should be apply the same logic as [here](https://github.com/nextflow-io/nextflow/blob/master/src/main/groovy/nextflow/processor/TaskProcessor.groovy#L1824-L1827). 

So I guess something like: 

```
def copy = [:]
fileMap.keySet().each { name -> 
  def val = fileMap[name]
  def normalized = normalizeInputToFiles(val,count)
  copy[name] = singleItemOrList(normalized)
}

task.setInput(param, copy)
```